service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/dataproccluster/startup.sh

```diff
@@ -0,0 +1,781 @@
+
+# Create a systemd service file for the ssh-agent
+cat << EOF >"${TERRA_SSH_AGENT_SERVICE}"
+[Unit]
+Description=Run an SSH agent for the Jupyter user
+
+[Service]
+ExecStart=${TERRA_SSH_AGENT_SCRIPT}
+User=${DATAPROC_USER}
+Restart=always
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+# Enable and start the startup service
+systemctl daemon-reload
+systemctl enable "${TERRA_SSH_AGENT_SERVICE_NAME}"
+systemctl start "${TERRA_SSH_AGENT_SERVICE_NAME}"
+
+# Set ssh-agent launch command in ~/.bashrc so everytime
+# user starts a shell, we start the ssh-agent.
+cat << EOF >> "${USER_BASHRC}"
+
+# Get the ssh-agent environment variables
+if [[ -f ~/.ssh-agent/environment ]]; then
+  eval "\$(<~/.ssh-agent/environment)" >/dev/null
+fi
+EOF
+
+#############################
+# Setup instance boot service
+#############################
+# Create a script to perform the following steps every time the instance boots:
+# 1. Mount terra workspace resources. This command requires system user home
+#    directories to be mounted. We run the startup service after
+#    jupyter.service to meet this requirement.
+
+emit "Setting up Terra boot script and service..."
+
+# Create the boot script
+cat << EOF >"${TERRA_BOOT_SCRIPT}"
+#!/bin/bash
+# This script is run on instance boot to configure the instance for terra.
+
+# Send stdout and stderr from this script to a file for debugging.
+exec >> "${TERRA_BOOT_SERVICE_OUTPUT_FILE}"
+exec 2>&1
+
+# Pick up environment from the ~/.bashrc
+source "${USER_BASHRC}"
+
+# Mount terra workspace resources
+"${USER_HOME_LOCAL_BIN}/terra" resource mount
+
+exit 0
+EOF
+chmod +x "${TERRA_BOOT_SCRIPT}"
+chown ${DATAPROC_USER}:${DATAPROC_USER} "${TERRA_BOOT_SCRIPT}"
+
+# Create a systemd service to run the boot script on system boot
+cat << EOF >"${TERRA_BOOT_SERVICE}"
+[Unit]
+Description=Configure environment for terra
+After=jupyter.service
+
+[Service]
+ExecStart=${TERRA_BOOT_SCRIPT}
+User=${DATAPROC_USER}
+RemainAfterExit=yes
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+# Enable and start the service
+systemctl daemon-reload
+systemctl enable "${TERRA_BOOT_SERVICE_NAME}"
+systemctl start "${TERRA_BOOT_SERVICE_NAME}"
+
+# Indicate the end of Terra customizations of the ~/.bashrc
+cat << EOF >> "${USER_BASHRC}"
+
+### END: Terra-specific customizations ###
+EOF
+
+# Make sure the ~/.bashrc and ~/.bash_profile are owned by the jupyter user
+chown ${DATAPROC_USER}:${DATAPROC_USER} "${USER_BASHRC}"
+chown ${DATAPROC_USER}:${DATAPROC_USER} "${USER_BASH_PROFILE}"
+
+
+###################################
+# Configure Jupyter systemd service
+###################################
+
+# By default the dataproc jupyter optional component runs jupyter as the root user. We override the behavior by configuring the jupyter service to run as the dataproc user instead.
+
+emit "Configuring Jupyter systemd service..."
+
+# Modify the jupyter service configuration
+cat << EOF >${JUPYTER_SERVICE}
```

Comment:
As mentioned above, I'm updating the jupyter service to run as the dataproc user. Also set the working directory to dataproc user's home folder.